### ISSUE TITLE
License?

### ISSUE THREAD

**Issue by cdeil** [ceyzeriat/astroobs#5]
I see that you've change the license from MIT to GPL for `astroobs`:

https://github.com/ceyzeriat/astroobs/commit/6a15824d283a7988ce57e9152260885837097d46#diff-d9ae695c8119e58c2ec435b6deae2d53R4

If you want to collaborate or even merge with efforts in Astropy and Astroplan, that's a problem.

Astropy, Astroplan and most Astropy-affiliated packages have a liberal license and we can't use or even read your GPL code, because using it for our BSD-licensed packages would be a violation of the GPL.

cc @bmorris3 and @eteq as Astroplan devs.

**Comment by ceyzeriat** [comment]
Hi there,
Indeed I changed the license lately because I always preferred the "more freedom" aspect of GPL over the "greater freedom" in BSD (and I should have actually changed that a long time ago).
I guess the only solution to merge is to slide astroobs back to BSD license, which I don't mind too much in the hope that astropy will remain maintained and open (and not fork to a pay-for-additional-features strategy).

I see however a difference in philosophy between astroplan and astroobs, in that astroobs tends to facilitate the user's work to the maximum (and I really mean maximum).
e.g.
Instead of writing these 3 lines with astroplan:

```
from astropy.time import Time
time = Time('2015-06-16 12:00:00')
subaru.target_is_up(time, altair)
```

one should (also) be able to write this single line:
`subaru.target_is_up([2015,6,16], 'altair')`
where 'altair' is a str hence queried automatically to CDS, and the time a simple int-list where the local midnight is assumed, i.e. [2015,6,16] is the night from 16th to 17th local time.

That's just a different philosophy, there is no right or wrong.
But then I guess the questions boils down to whether you guys are ok to take care of lazy users? ;)
Guillaume

**Comment by cdeil** [comment]
OK, so you prefer GPL.

Probably we should just table the license discussion in this issue and first talk about whether collaboration or merging is desired or useful.

> That's just a different philosophy, there is no right or wrong.
> But then I guess the questions boils down to whether you guys are ok to take care of lazy users? ;)

I don't know.

There's probably a few cases where you don't know the simplest way to do something with Astropy / Astroplan (no problem, your examples / feedback will lead to doc improvements) and a few cases where the Astroplan API can be simplified (e.g. taking time strings as input in addition to Time objects, as long as the Time constructor can parse them) and then there will be some cases where some Astropy / Astroplan people think more verbose / explicit is better, i.e. real differences in API philosophy.

An issue pointing out things you think are too cumbersome with Astroplan at the moment (and maybe showing the astroobs API for those examples) would be welcome for sure and lead to a better understanding of what the differences between the packages are.

**Comment by ceyzeriat** [comment]
Both packages have their strengths:
- astroplan is astropy powered, does not need pyephem to run,
- astroobs is dumb-easy to use and provides outputs that doesn't require any prior knowledge of astropy, and even python.

And weaknesses:
- astroplan is, to me, cumbersome to use, which defeats the purpose of an observation planning tool (many observing runs don't go as planned and back-up targets need to be found quickly),
- astroobs is not astropy powered, uses pyephem (quite slow on some calculations)

They would both gain in merging if one takes care of keeping strengths of each package, and cancel-out the weaknesses.
Now this is my view, and I don't want to push people. I am happy to spend some time pointing out where the cumbersomeness is in astroplan, but I am not sure that it will be time spent very wisely if one of the fundamentals of astroplan is to be fully explicit. One can't be both explicit and flexible on inputs.
To sum up my "ease of use" specifications, a user should be able to plot an elevation curve of a couple of targets+moon at a given date and observatory, in 4-5 lines including imports, and without knowing a-priori any formatting other than basic python.
If that fits with astroplan line, then lets merge!
